### PR TITLE
[Dependency Update]Update the numpy version

### PR DESCRIPTION
**Pull request by stu1130**
## Description ##
* Numpy version less and equal to 1.16.0 has been found with security vulnerability. Detailed can be find in this link: https://nvd.nist.gov/vuln/detail/CVE-2019-6446#vulnCurrentDescriptionTitle.
Upgrade the numpy version both in setup.py and ci part to mitigate security vulnerability
* Update the dmlc-core commit to use python3 linter

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

## Comments ##
disable several pylint comparison-with-callable on event_handler @roywei please check it

### CODE REVIEW

**Comment by piyushghai** [comment]
@stu1130 Seems like not all the CI passed on this PR. 
Would you be taking a look into the failures ?

@mxnet-label-bot Add [Numpy, pr-work-in-progress]

**Comment by Roshrini** [comment]
@stu1130 Can you take a look at CI failure.

**Comment by stu1130** [comment]
@szha the PR is ready for review

**Comment by stu1130** [comment]
@mxnet-label-bot Update [Numpy, pr-awaiting-review]

**Comment by stu1130** [comment]
ping @szha for review thanks

File: tools/pip/setup.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def has_ext_modules(self):
 
 
 DEPENDENCIES = [
-    'numpy<1.15.0,>=1.8.2',
+    'numpy>1.16.0',
```
Comment (szha) [inline tools/pip/setup.py]: please add upperbound. (e.g. major version change)

**Pull request merged**